### ISSUE TITLE
Allow Qt applications to work without QT_X11_NO_MITSHM

### ISSUE THREAD

**Issue by alexpattyn** [debarshiray/toolbox#163]
# Issue

I would like to use toolbox for when I am programing. Currently I am using julia (https://julialang.org/) when I want to plot something like...

```julia
using Plots

t = 0:0.1:2pi;
plot(t,sin.(t))
```
I got errors that complained about missing qt libraries, so I used  `sudo dnf install qt-devel`. Which then allowed me to draw a window with the above code but with the following error (also it was just an empty window without a plot)...
```julia
julia> X Error: BadAccess (attempt to access private resource denied) 10
  Extension:    130 (MIT-SHM)
  Minor opcode: 1 (X_ShmAttach)
  Resource id:  0x1600002
X Error: BadShmSeg (invalid shared segment parameter) 128
  Extension:    130 (MIT-SHM)
  Minor opcode: 3 (X_ShmPutImage)
  Resource id:  0x160000e
```

**Comment by koenigjaeger** [comment]
Maybe related to https://github.com/debarshiray/toolbox/issues/104? Although I don't get any warnings about environmental variables.

**Comment by debarshiray** [comment]
I get another error message before those X errors:
```
QGtkStyle could not resolve GTK. Make sure you have installed the proper libraries.
```

Searching for that message on the Internet makes me think that this is related to Qt trying to use a theme and failing. I don't know why. Maybe @grulja has an idea?

In my case I had `qt-4.8.7-45.fc29.x86_64`. I wonder if Qt 5 applications have this problem.

**Comment by grulja** [comment]
It tries to load gtk-x11-2.0 library and seem to fail. Qt 5 applications will not have this problem, because this gtk style is not part of Qt anymore. The old style was using Gtk calls directly to render widgets, we now use a completely different approach with adwaita-qt theme.

**Comment by debarshiray** [comment]
@grulja I see, ok. Installing `gtk2` got me past that error.

As for the original X error, one workaround is to set `QT_X11_NO_MITSHM=1` in the environment:
```
$ QT_X11_NO_MITSHM=1 julia
```

**Comment by debarshiray** [comment]
> As for the original X error, one workaround is to set `QT_X11_NO_MITSHM=1` in the environment:

Using the host's IPC namespace (eg., /dev/shm) makes the problem go away.

**Comment by debarshiray** [comment]
Re-create your container with `toolbox.git` to solve this problem. Thanks for stopping by.

**Comment by alexpattyn** [comment]
Thanks! Is the fix in `0.0.10-1.fc30` that is currently in testing?

**Comment by debarshiray** [comment]
> Thanks! Is the fix in `0.0.10-1.fc30` that is currently in testing?

Sadly, no; but there will be another release in another ten days or so.
